src/frontend/components/DashboardVideoLivePairing/index.tsx

```diff
@@ -0,0 +1,139 @@
+    description:
+      'Message displayed when the secret for pairing an external device has expired.',
+    id: 'components.DashboardVideoLivePairing.pairingSecretCountdownExpired',
+  },
+});
+
+interface DashboardVideoLivePairingProps {
+  video: Video;
+}
+
+export const DashboardVideoLivePairing = ({
+  video,
+}: DashboardVideoLivePairingProps) => {
+  const [buttonColor, setButtonColor] = useState('brand');
+  const [secret, setSecret] = useState('');
+  const [expiresIn, setExpiresIn] = useState(0);
+  const [secondsLeft, setSecondsLeft] = useState(0);
+  const [percentsLeft, setPercentsLeft] = useState(100);
+  const [expired, setExpired] = useState(false);
+
+  const usePairingVideoMutation = usePairingVideo(video.id, {
+    onSuccess: (data) => {
+      setSecret(data.secret);
+      setExpiresIn(data.expires_in);
+      setSecondsLeft(data.expires_in);
+      setPercentsLeft(100);
+      setExpired(false);
+      setButtonColor('status-ok');
+    },
+  });
+
+  const pairVideoAction = () => {
+    usePairingVideoMutation.mutate();
+  };
+
+  const onInterval = () => {
+    setSecondsLeft(secondsLeft - 1);
+    setPercentsLeft((secondsLeft * 100) / expiresIn);
+
+    if (secondsLeft <= 0) {
+      if (secondsLeft > -3) {
+        setExpired(true);
+        setButtonColor('status-warning');
+      } else {
+        setSecret('');
+        setButtonColor('brand');
+      }
+    }
+  };
+
+  useEffect(() => {
+    let interval: number | undefined;
+    if (secret) {
+      interval = window.setInterval(onInterval, 1000);
+    }
+    if (secondsLeft < -4) {
+      window.clearInterval(interval);
+    }
+
+    return () => {
+      window.clearInterval(interval);
+    };
+  }, [secondsLeft]);
+
+  let content: JSX.Element;
+
+  if (!secret) {
+    // no secret yet, initial state
+    content = (
+      <Text alignSelf="center">
+        <FormattedMessage {...messages.pairingSecretLabel} />
+      </Text>
+    );
+  } else {
+    if (!expired) {
+      // secret exists and is not expired yet
+      content = (
+        <Box>
+          <Text alignSelf="center" size="large">
+            <FormattedMessage
+              {...messages.pairingSecretDisplay}
+              values={{ secret }}
+            />
+          </Text>
+          <Meter
+            type="bar"
+            color="light-1"
+            background={{ color: 'light-1', opacity: 'medium' }}
+            value={percentsLeft}
+            alignSelf="center"
+            size="full"
+            thickness="xsmall"
+            margin={{ top: 'small' }}
+          />
+        </Box>
+      );
+    } else {
+      // secret has expired
+      content = (
+        <Text alignSelf="center">
+          <FormattedMessage {...messages.pairingSecretCountdownExpired} />
+        </Text>
+      );
+    }
+  }
+
+  return (
+    <DashboardButton
+      primary={true}
+      color={buttonColor}
+      onClick={pairVideoAction}
+    >
+      <Box alignSelf="center">{content}</Box>
```

Comment:
Adding a `Box` (so a `div`) in abutton is invalid. You can't do that.

Reply:
Good catch, I applied this: https://storybook.grommet.io/?path=/story/controls-button-children--children
Looking at the design, this will change anyway…

Reply:
FYI @lunika , I'm looking for another way to build it.